models/action_recognition/model_templates/gesture-recognition/README.md

```diff
@@ -36,78 +35,98 @@ Activate virtual environment:
 ```
 
-> **NOTE**: It's recommended during fine-tuning to decrease the `--base-learning-rate` parameter compared with default value (see `${MODEL_TEMPLATE}`) to prevent from forgetting during the first iterations.
+### 4. Fine-tune
 
-Also you can use parameters such as `--epochs`, `--batch-size`, `--gpu-num`, `--base-learning-rate`, otherwise default values will be loaded from `${MODEL_TEMPLATE}`.
+#### a. Prepare dataset
 
-### 6. Evaluation
+Prepare one of the listed below datasets for training or collect and annotate your own:
+* To prepare MS-ASL data follow instructions: [DATA_MSASL.md](./DATA_MSASL.md).
+* To prepare JESTER data follow instructions: [DATA_JESTER.md](./DATA_JESTER.md).
 
-Evaluation procedure allows us to get quality metrics values and complexity numbers such as number of parameters and FLOPs.
+Set some environment variables:
+```bash
+export ADD_EPOCHS=1
+export EPOCHS_NUM=$((`cat ${MODEL_TEMPLATE} | grep epochs | tr -dc '0-9'` + ${ADD_EPOCHS}))
+```
 
-To compute mean accuracy metric run:
+#### b. Evaluate
 
 ```bash
 python eval.py \
-   --load-weights ${WORK_DIR}/outputs/latest.pth \
+   --load-weights ${SNAPSHOT} \
    --test-ann-files ${TEST_ANN_FILE} \
    --test-data-roots ${TEST_DATA_ROOT} \
-   --save-metrics-to ${WORK_DIR}/metrics.yaml
+   --save-metrics-to metrics.yaml
 ```
 
-### 7. Export PyTorch\* model to the OpenVINO™ format
+If you would like to evaluate exported model, you need to pass `export/model.bin` instead of passing `${SNAPSHOT}` .
 
-To convert PyTorch\* model to the OpenVINO™ IR format run the `export.py` script:
+#### c. Fine-tune or train from scratch
 
-```bash
-python export.py \
-   --load-weights ${WORK_DIR}/outputs/latest.pth \
-   --save-model-to ${WORK_DIR}/export
-```
+Try both following variants and select the best one:
 
-This produces model `model.xml` and weights `model.bin` in single-precision floating-point format
-(FP32). The obtained model expects **normalized image** in planar RGB format.
+   * **Fine-tuning** from pre-trained weights. If the dataset is not big enough, then the model tends to overfit quickly, forgetting about the data that was used for pre-training and reducing the generalization ability of the final model. Hence, small starting learning rate and short training schedule are recommended.
+   * **Training** from scratch or pre-trained weights. Only if you have a lot of data, let's say tens of thousands or even more images. This variant assumes long training process starting from big values of learning rate and eventually decreasing it according to a training schedule.
+
+   * If you would like to start **fine-tuning** from pre-trained weights use `--resume-from` parameter and value of `--epochs` have to exceed the value stored inside `${MODEL_TEMPLATE}` file, otherwise training will be ended immediately. Here we add `1` additional epoch.
+
+      ```bash
+      python train.py \
+         --resume-from ${SNAPSHOT} \
```

Comment:
I'm sorry but `resume-from` will not work if the number of classes is different from `snapshot.pth`. One more objection to use `resume-from` is related to the ratio of leaning speed between new parameters (in head) and old (in backbone) - I'm not sure that such strategy will survive without `warm-up`. I would prefer to use `load-from` always (as it was originally) to guarantee that learning walked through the whole pipeline (which includes `warm-up` too). In light of this I select only two stages: fine-tuning (training from snapshot) and training from scratch.

Reply:
it is sad, since the `--resume-from` is not supported with provided pre-trained weights and user has no option to run full-time training or just fine-tune during a couple of epochs. Please have a look at https://github.com/openvinotoolkit/training_extensions/pull/480/commits/b641eb53d5ac86235f308a5ba9df9a34c0fe1cc9 and let me know if it satisfies you.

Reply:
I would prefer to exclude the `--resume-from` option completely and change names: "training" - training from scratch (w/o specifying `snapshot.pth`) and "fine-tuning" - training with `--load-from ${SNAPSHOT}` option. Regarding "short" / "full" fine-tuning mode - we can add `note` section ans specify there that user can set `--epoch` at any value (`> 10 epochs` to be sure that 'warm-up' is finished) and `--base-learning-rate` to `1e-3` to achieve "short" schedule.

Reply:
It will make alignment across algos in even less degree (if it is done for gesture recognition only). If decide to completely remove `--resume-from` it will require much more changes including parsers, converters and so on. This is divisible but not within this PR. More schedules might confuse user. 

So I would postpone this step.

Reply:
@evgeny-izutov ?

Reply:
“Not everything that is faced can be changed, but nothing can be changed until it is faced.” -- Ok.